ddnoise.js

```diff
@@ -64,11 +64,10 @@ define(['./utils', 'underscore', 'promise'], function (utils, _) {
     };
 
     DdNoise.prototype.play = function (sound, loop) {
+        if (this.context.state !== "running") return Promise.reject();
         var self = this;
-        var context = self.context;
-        if (context.state !== "running") return null;
         return new Promise(function (resolve, reject) {
-            var source = context.createBufferSource();
+            var source = self.context.createBufferSource();
```

Comment:
Oh!!! I was purely being blind before: feel free to put it back as it was! Sorry :/

Reply:
I think I'll keep the latest -- in terms of clarity it feels like the best of both worlds. Now, the inner function uses self exclusively for outer function access, and the outer function uses this exclusively.